include/nbl/asset/ICPURayTracingPipeline.h

```diff
@@ -0,0 +1,128 @@
+
+// Copyright (C) 2018-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_ASSET_I_CPU_RAY_TRACING_PIPELINE_H_INCLUDED_
+#define _NBL_ASSET_I_CPU_RAY_TRACING_PIPELINE_H_INCLUDED_
+
+#include "nbl/asset/IRayTracingPipeline.h"
+#include "nbl/asset/ICPUPipeline.h"
+
+
+namespace nbl::asset
+{
+
+//! CPU Version of RayTracing Pipeline
+class ICPURayTracingPipeline final : public ICPUPipeline<IRayTracingPipeline<ICPUPipelineLayout>>
+{
+        using pipeline_base_t = IRayTracingPipeline<ICPUPipelineLayout>;
+        using base_t = ICPUPipeline<pipeline_base_t>;
+
+    public:
+        struct SHitGroupSpecInfos {
+            core::vector<SShaderSpecInfo> closestHits;
+            core::vector<SShaderSpecInfo> anyHits;
+            core::vector<SShaderSpecInfo> intersections;
```

Comment:
I'd prefer AoS instead of SoA, why have you got them laid out this way so the coupling og the group isn't apparent (which shaders go together)?

Reply:
aah okay, because of the spans to each stage separately. I get it now